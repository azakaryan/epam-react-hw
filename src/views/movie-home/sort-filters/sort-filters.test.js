import React from 'react';
import { shallow } from 'enzyme';
import expectEnzymify from 'expect';
import enzymify from 'expect-enzyme';
import SortFilters from './sort-filters';

expectEnzymify.extend(enzymify());
const onFiltersApply = () => {};

describe('Sort Filters Component', () => {
  it('should render correctly', () => {
    const component = shallow(<SortFilters onFiltersApply={onFiltersApply} />);

    expect(component).toMatchSnapshot();
  });

  it('renders a sort filters component', () => {
    const wrapper = shallow(<SortFilters onFiltersApply={onFiltersApply} />);
    expectEnzymify(wrapper).toBeA('div');
  });

  it('set state on sort filters component', () => {
    const wrapper = shallow(<SortFilters onFiltersApply={onFiltersApply} />);
    wrapper.find({ id: 'rating' }).simulate('click');
    expect(wrapper.state('sortByValue')).toEqual('vote_count');
    wrapper.find({ id: 'release_date' }).simulate('click');
    expect(wrapper.state('sortByValue')).toEqual('release_date');
  });

  it('check default state to be equal to "release_date" on sort filters component', () => {
    const wrapper = shallow(<SortFilters onFiltersApply={onFiltersApply} />);
    expect(wrapper.state('sortByValue')).toEqual('release_date');
  });

  it('simulate filter click on sort filters component', () => {
    let k = 1;
    const onFiltersApplyFunc = () => k = k + 1;
    const wrapper = shallow(<SortFilters onFiltersApply={onFiltersApplyFunc} />);
    expect(k).toEqual(2);
    wrapper.find({ id: 'release_date' }).simulate('click');
    expect(k).toEqual(3);
  });
});
