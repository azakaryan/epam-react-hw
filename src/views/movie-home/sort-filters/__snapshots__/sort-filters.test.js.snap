// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort Filters Component should render correctly 1`] = `
<div
  className="main_container"
>
  <div
    className="title"
  >
    SORT BY
  </div>
  <div
    className="content"
  >
    <label
      className="active"
      id="release_date"
      onClick={[Function]}
    >
      release date
    </label>
    <label
      className=""
      id="rating"
      onClick={[Function]}
    >
      rating
    </label>
  </div>
</div>
`;
